ApiDoctor.Console/Program.cs

```diff
@@ -1960,14 +1957,149 @@ private static async Task<bool> GenerateSnippetsAsync(GenerateSnippetsOptions op
+                snippetsTabSectionForMethod.Append("---\r\n"); // append end of tab section
+
+            // Dump the SDK link file if doesn't exist
+            var sdkFileFullName = Path.Combine(docsSnippetsDirectory, includeSdkFileName);
+            if (!File.Exists(sdkFileFullName))
+            {
+                const string includeSdkText = "<!-- markdownlint-disable MD041-->\r\n\r\n" +
+                    "> Read the [SDK documentation](https://docs.microsoft.com/graph/sdks/sdks-overview) " +
+                    "for details on how to [add the SDK](https://docs.microsoft.com/graph/sdks/sdk-installation) to your project and " +
+                    "[create an authProvider](https://docs.microsoft.com/graph/sdks/choose-authentication-providers) instance.";
+                File.WriteAllText(sdkFileFullName, includeSdkText);
+            }
+
+            // Dump the snippet not available file if it does not exist
+            var snippetNotAvailableFileFullName = Path.Combine(docsSnippetsDirectory, includeSnippetsNotAvailableFileName);
+            if (!File.Exists(snippetNotAvailableFileFullName))
+            {
+                const string includeSnippetNotAvailableText = "```\r\nSnippet not available\r\n```";
+                File.WriteAllText(snippetNotAvailableFileFullName, includeSnippetNotAvailableText);
+            }
+            return snippetsTabSectionForMethod.ToString();
+        }
+
+        private static string GetSnippetContentForMethod(string language, string snippetPrefix, string snippetsPath)
+        {
+            var fileName = $"{snippetPrefix}---{language.ToLowerInvariant()}";
+            var fileFullPath = Path.Combine(snippetsPath, fileName);
+
+            FancyConsole.WriteLine(FancyConsole.ConsoleSuccessColor, $"Reading {fileFullPath}");
+
+            string codeSnippet = null;
+            if (File.Exists(fileFullPath))
+                codeSnippet = File.ReadAllText(fileFullPath);
+            else
+                FancyConsole.WriteLine(FancyConsole.ConsoleErrorColor, $"Error: file '{fileName}' does not exist");
+
+            return codeSnippet;
+        }
+
+        private static Dictionary<string, HashSet<string>> SnippetsLanguageSetForDocument = new Dictionary<string, HashSet<string>>();
+
+        private static HashSet<string> GetSnippetsLanguagesToIncludeForMethod(MethodDefinition sourceMethod, MethodDefinition[] methodCollection, string[] languageOptions, string snippetsPath, string snippetPrefix)
+        {
+            var languages = LookupGeneratedSnippetsLanguagesBySnippetPrefix(snippetPrefix, snippetsPath);
+            if (!languages.Any())
+            {
+                FancyConsole.WriteLine(FancyConsole.ConsoleWarningColor, $"Failed to fetch snippets for method: {sourceMethod.Identifier}");
+                return new HashSet<string>();
+            }
+            else if (SnippetsLanguageSetForDocument.TryGetValue(sourceMethod.SourceFile.DisplayName, out var storedLanguages))
```

Comment:
Maybe we should check this first before calling `LookupGeneratedSnippetsLanguagesBySnippetPrefix` in line 2050. Since it we will have done wasteful work if we enter this block.

Reply:
I agree. The aim was to not return any languages for methods that do not have any code snippets even though the page has other methods with at least one snippet. There's a more efficient way to do it.